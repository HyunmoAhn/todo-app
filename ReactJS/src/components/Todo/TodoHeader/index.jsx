import React from 'react';
// import PropTypes from 'prop-types';

const propTypes = {};
const defaultProps = {};

class TodoHeader extends React.Component {
  render() {
    return (
      <header className="header">
        <h1>todos</h1>
        <input className="new-todo" type="text" placeholder="What needs to be done?" />
        <input className="toggle-all" id="toggle-all" type="checkbox" />
        <label style={{ top: '15px' }} htmlFor="toggle-all">Mark all async complete</label>
      </header>
    );
  }
}

TodoHeader.propTypes = propTypes;
TodoHeader.defaultProps = defaultProps;

export default TodoHeader;
