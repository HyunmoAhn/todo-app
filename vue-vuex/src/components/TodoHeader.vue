<template>
  <header class="header">
    <h1>todos</h1>
    <input
      class="new-todo"
      type="text"
      v-model.trim="value"
      @keydown.enter="addProcess"
      placeholder="What needs to be done?"
    />
  </header>
</template>
<script>
import { mapMutations } from 'vuex';
import { v4 } from 'uuid';
import * as mutations from '@/store/mutations';

export default {
  data() {
    return {
      value: '',
    };
  },
  methods: {
    ...mapMutations([
      mutations.ADD_TODO,
    ]),
    addProcess() {
      if (this.value) {
        this[mutations.ADD_TODO]({ id: v4(), content: this.value });
        this.value = '';
      }
    },
  },
};
</script>
