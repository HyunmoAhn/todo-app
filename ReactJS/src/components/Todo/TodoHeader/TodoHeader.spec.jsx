import React from 'react';
import { shallow } from 'enzyme';

import TodoHeader from './';

describe('<TodoHeader />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<TodoHeader />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should change state when call handleChange', () => {
    const event = {
      target: {
        value: 'mock-value',
      },
    };
    const wrapper = shallow(<TodoHeader />);
    wrapper.setState({
      value: 'initial-value',
    });

    wrapper.instance().handleChange(event);

    expect(wrapper.state().value).toBe('mock-value');
  });

  describe('test handleInputKeyPress', () => {
    const handleAddList = jest.spyOn(TodoHeader.prototype, 'handleAddList');

    beforeEach(() => {
      jest.clearAllMocks();
    });

    it('when press enter key', () => {
      const event = {
        key: 'Enter',
      };
      const wrapper = shallow(<TodoHeader />);
      wrapper.setState({
        value: 'initial-state',
      });

      wrapper.instance().handleInputKeyPress(event);

      expect(handleAddList).toHaveBeenCalledTimes(1);
      expect(handleAddList).toHaveBeenCalledWith();
      expect(wrapper.state().value).toBe('');
    });

    it('when press except enter key', () => {
      const event = {
        key: 'not-enter',
      };
      const wrapper = shallow(<TodoHeader />);
      wrapper.setState({
        value: 'initial-state',
      });

      wrapper.instance().handleInputKeyPress(event);

      expect(handleAddList).toHaveBeenCalledTimes(0);
      expect(wrapper.state().value).toBe('initial-state');
    });
  });

  describe('test handleAddList', () => {
    it('when state.value is existed', () => {
      const onAddListItem = jest.fn();
      const wrapper = shallow(<TodoHeader onAddListItem={onAddListItem} />);
      wrapper.setState({
        value: 'exist',
      });

      wrapper.instance().handleAddList();

      expect(onAddListItem).toHaveBeenCalledTimes(1);
      expect(onAddListItem).toHaveBeenCalledWith('exist');
    });


    it('when state.value is existed', () => {
      const onAddListItem = jest.fn();
      const wrapper = shallow(<TodoHeader onAddListItem={onAddListItem} />);
      wrapper.setState({
        value: '  ',
      });

      wrapper.instance().handleAddList();

      expect(onAddListItem).toHaveBeenCalledTimes(0);
    });
  });
});
