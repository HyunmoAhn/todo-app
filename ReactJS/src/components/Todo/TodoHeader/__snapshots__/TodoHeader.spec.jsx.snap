// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoHeader /> should match to snapshot when render default 1`] = `
<header
  className="header"
>
  <h1>
    todos
  </h1>
  <input
    className="new-todo"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
  <input
    checked={false}
    className="toggle-all"
    id="toggle-all"
    onClick={[Function]}
    readOnly={true}
    type="checkbox"
  />
  <label
    htmlFor="toggle-all"
    style={
      Object {
        "top": "15px",
      }
    }
  >
    Mark all async complete
  </label>
</header>
`;
